pandas/core/algorithms.py

```diff
@@ -219,6 +219,10 @@ def _get_data_algo(values, func_map):
     if is_categorical_dtype(values):
         values = values._values_for_rank()
 
+    # Create copy in case NaT converts to asi8
```

Comment:
I wouldn't do this here at all, this will affect virtually all algos. in ``pandas/_libs/algos_rank_helper.pxi.in``, you can just do

```
if mask.any():
    values = values.copy()
    np.putmask(....)
```

Reply:
Thanks, it makes a lot more sense now and exercises a gentle introduction to working with cython for me!